feat: introduce new centralized OTLP gateway architecture (#3148)
kyma-project/telemetry-manager: docs/user/collecting-traces/README.md

Reviewer: ```suggestion
- gRPC: `http://telemetry-otlp.kyma-system:4317`
```

Reviewer: ```suggestion
- HTTP: `http://telemetry-otlp.kyma-system:4318`
```

## Overview

A TracePipeline is a Kubernetes custom resource (CR) that configures trace collection for your cluster. When you create a TracePipeline, the OTLP Gateway provides a central OTLP endpoint receiving traces pushed from applications (for details, see [Traces Architecture](./../architecture/traces-architecture.md)).

The pipeline enriches all collected traces with Kubernetes metadata before sending them to your chosen backend.

> Trace collection is optional. If no pipeline resource exists, the OTLP Gateway is not deployed.

Reviewer: That's an interesting situation. 
We must assume that the OTLP Gateway basically always exists (Telemetry module without a single pipeline is useless), but STILL traces are only collected if the user defines a TracePipeline (regardless of the existence of an OTLP Gateway).

I believe what we're trying to say here is something like this:

```suggestion
Trace collection is optional. If you don't create a TracePipeline, you don't collect traces.
```

or

```suggestion
Trace collection is optional. If you don't want to collect traces, don't create a TracePipeline.
```

(the user probably doesn't care about the existence of components, but about what effects it has on their resource consumption etc.)
